authors/apps/social_auth/login_register.py

```diff
@@ -0,0 +1,32 @@
+from django.core.exceptions import ValidationError
+
+from authors.apps.authentication.models import User
+
+
+def login_or_register_social_user(social_user):
+    try:
+        user = User.objects.get(email=social_user.get('email'))
+        return {
+            "email": user.email,
+            "username": user.username,
+            "token": user.auth_token()
+        }
+    except User.DoesNotExist:
+        if social_user.get("email") is None:
+            raise ValidationError("Users must have an email address.")
+
+        new_social_user = {
+            "email": social_user.get("email"),
+            "username": social_user.get("name", "unknown"),
+            "password": User.objects.make_random_password()
+        }
+
+        new_user = User.objects.create_user(**new_social_user)
+        new_user.is_active = True
+        new_user.save()
+
+        return {
```

Comment:
This seems to be repeated with code on line 9.
you can find a way to write it one

Reply:
Thank you for the feedback. This has been implemented.